[TASK] Revise chapter "Page title API" (#2336)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/Seo/PageTitleApi.rst

Define priority of PageTitleProviders
=====================================

> The priority of the providers are set by the TypoScript property

Reviewer: ```suggestion
The priority of the providers are set by the TypoScript property
```
```suggestion
The priority of the providers is set by the TypoScript property
```

:typoscript:`config.pageTitleProviders`. This way an integrator is able to set
the priorities for his project and can even have conditions in place.

By default, the Core has the following setup:

..  code-block:: typoscript

    config.pageTitleProviders {
        record {
            provider = TYPO3\CMS\Core\PageTitle\RecordPageTitleProvider
        }
    }

The sorting of the providers is based on the :typoscript:`before` and
:typoscript:`after` parameters. If you want a provider to be handled before a
specific other provider, just set that provider in the :typoscript:`before`,
do the same with :typoscript:`after`.

If you have installed the system extension SEO, you will also get a second
provider. The configuration will be:

..  code-block:: typoscript

    config.pageTitleProviders {
        record {
            provider = TYPO3\CMS\Core\PageTitle\RecordPageTitleProvider
        }
        seo {
            provider = TYPO3\CMS\Seo\PageTitle\SeoTitlePageTitleProvider
            before = record
        }
    }

First the :php:`SeoTitlePageTitleProvider` (because it will be handled before

> record) and, if this providers didn't provide a title, the

Reviewer: ```suggestion
:typoscript:`record`) and, if this providers didn't provide a title, the
```

> record) and, if this providers didn't provide a title, the

Reviewer: ```suggestion
record) and, if this providers did not provide a title, the
```

> record) and, if this providers did not provide a title, the

Author: ```suggestion
`typoscript`:record:) and, if this providers did not provide a title, the
```

:php:`RecordPageTitleProvider` will be checked.

You can override these settings within your own installation. You can add as
many providers as you want. Be aware that if a provider returns a non-empty

> value, all provider with a lower priority won't be checked.

Reviewer: ```suggestion
value, all provider with a lower priority won't be checked.
```
```suggestion
value, all provider with a lower priority will not be checked.
```
